src/ray/core_worker/reference_count.cc

```diff
@@ -818,6 +818,28 @@ void ReferenceCounter::AddNestedObjectIds(const ObjectID &object_id,
   AddNestedObjectIdsInternal(object_id, inner_ids, owner_address);
 }
 
+void ReferenceCounter::UpdateOwnedObjectNestedObjectIds(
```

Comment:
There is an `AddNestedObjectIds` method that we should probably modify/use instead.

Also, we need to pass the inner IDs' owner addresses here.